lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/Explanation.java

```diff
@@ -1,111 +1,7 @@
-
-    public Explanation(List<AttributeSet> resultList,
-                       long numInliers,
-                       long numOutliers,
-                       long creationTimeMs) {
-        itemsets = new ArrayList<>(resultList);
-        this.numInliers = numInliers;
-        this.numOutliers = numOutliers;
-        this.creationTimeMs = creationTimeMs;
-    }
-
-    /**
-     * Removes redundant explanations
-     * @return New explanation with redundant itemsets removed.
-     */
-    public Explanation prune() {
-        itemsets.sort((AttributeSet a, AttributeSet b) -> -a.compareTo(b));
-        List<AttributeSet> newItemsets = new ArrayList<>();
-        int n = itemsets.size();
-        for (int i = 0; i < n; i++) {
-            AttributeSet aSet = itemsets.get(i);
-            boolean redundant = false;
-            // an explanation is redundant if it has lower risk ratio (occurs after since sorted)
-            // than an explanation that involves a subset of the same attributes
-            for (int j = 0; j < i; j++) {
-                AttributeSet comparisonSet = itemsets.get(j);
-                if (aSet.contains(comparisonSet)) {
-                    redundant = true;
-                    break;
-                }
-            }
-            if (!redundant) {
-                newItemsets.add(aSet);
-            }
-        }
-
-        Explanation newExplanation = new Explanation(
-                newItemsets,
-                numInliers,
-                numOutliers,
-                creationTimeMs
-        );
-        return newExplanation;
-    }
-
-    public void sortByRiskRatio() {
-        itemsets.sort((AttributeSet a, AttributeSet b) -> -a.compareTo(b));
-    }
-
-    public void sortBySupport() {
-        itemsets.sort((AttributeSet a, AttributeSet b) -> -Double.compare(a.getSupport(),b.getSupport()));
-    }
-
-    public List<AttributeSet> getItemsets() {
-        return itemsets;
-    }
-
-    public long getNumOutliers() {
-        return numOutliers;
-    }
-
-    public long getNumInliers() {
-        return numInliers;
-    }
-
-    public long getCreationTimeMs() {
-        return creationTimeMs;
-    }
-
-    public String prettyPrint() {
-        StringBuilder header = new StringBuilder(String.format(
-                "Outlier Explanation:\n"
-                + "numOutliers: %d\n"
-                + "numInliers: %d\n"
-                + "Itemsets: \n"
-                + "--------\n",
-                numOutliers,
-                numInliers));
-        for (AttributeSet is : itemsets) {
-            header.append(is.prettyPrint());
-        }
-        return header.toString();
-    }
-
-    @Override
-    public String toString() {
-        return "Explanation{" +
-                "numOutliers=" + numOutliers +
-                ", numInliers=" + numInliers +
-                ", itemsets=" + itemsets +
-                '}';
-    }
+public interface Explanation {
+    String prettyPrint();
```

Comment:
Which part are you referring to? Defining interfaces like this is a pretty common way to support both old and new code.